### PR TITLE
Add CartSummary Component

### PR DESCRIPTION
**Pull request by willopez**
Adds a `CartSummary` component to renders a summary of the current items in the cart.

### CODE REVIEW

File: package/src/components/CartSummary/v1/CartSummary.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { applyTheme } from "../../../utils";
+
+const Table = styled.table`
+  min-width: 20rem;
```
Comment (nnnnat) [inline package/src/components/CartSummary/v1/CartSummary.js]: This `min-width` is causing the table to overflow its container on smaller mobile devices. I think you can get a similar layout by removing the `min-width` and adding `width: 100%` and possibly a `max-width` if we need it to cap its width but from the looks of the designs, I think this component should just fill the available space.

File: package/src/components/CartSummary/v1/CartSummary.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+### CartSummary
+
+### Overview
+Displays a summary of the current current items in the cart.
+
+#### Basic usage
+```jsx
+<CartSummary
+  displayShipping="10.99"
```
Comment (nnnnat) [inline package/src/components/CartSummary/v1/CartSummary.md]: Missing $ for subtotal, shipping, and total

File: package/src/components/CartSummary/v1/CartSummary.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  displaySubtotal="275.77"
+  displayTotal="286.10"
+  itemsQuantity={3}
+/>
+```
+
+#### Renders summary with FREE shipping
+```jsx
+<CartSummary
+  displayShipping="3-5 Days"
```
Comment (nnnnat) [inline package/src/components/CartSummary/v1/CartSummary.md]: Is the `displayShipping` prop needed if `isFreeShipping` is true?

File: package/src/components/CartSummary/v1/CartSummary.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  displaySubtotal="275.77"
```
Comment (nnnnat) [inline package/src/components/CartSummary/v1/CartSummary.md]: Missing $ for subTotal and total

File: package/src/components/CartSummary/v1/CartSummary.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  itemsQuantity={3}
+  isFreeShipping
+/>
+```
+
+#### Renders summary with calculated taxes
+```jsx
+<CartSummary
+  displayShipping="$5.25"
+  displaySubtotal="275.77"
```
Comment (nnnnat) [inline package/src/components/CartSummary/v1/CartSummary.md]: Missing $ for subtotal and total

File: package/src/components/CartSummary/v1/CartSummary.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  displayTax="$7.62"
+/>
+```
+
+#### Renders summary with applied discount
+```jsx
+<CartSummary
+  displayDiscount="-$83.42"
+  displayShipping="$5.25"
+  displaySubtotal="275.77"
```
Comment (nnnnat) [inline package/src/components/CartSummary/v1/CartSummary.md]: Missing $ for subtotal and total

**Comment by willopez** [comment]
@nnnnat comments addressed

**Pull request merged**